### ISSUE TITLE
schedule of xent-regularize

### ISSUE THREAD

**Issue by danpovey** [kaldi-asr/kaldi#1341]
@hhadian, perhaps you could help with this...
we normally set xent-regularize in chain-training to a constant (e.g. 0.1).  But it seems to me that it might help to start at a larger value.  So you could call the python script with
 ` --xent-regularize=0.2,0.1`
and it would interpolate between 0.2 at the start and 0.1 at the end.  [note: it would be a string-type variable at the python-script level.
I'd like to have it vary with the num-archives processed and not with the iteration, which will make the training a little more invariant to the num-jobs-initial and num-jobs-final.  You can see the recent PR where Vimal implemented the dropout schedule, as an example, although I expect the code for this change would be considerably simpler.

**Comment by hhadian** [comment]
Ok will do it.
What would you think about making it more generic so we could set `--xent-regularize="20%:0.2, 80%:0.1"` so that only the first 20% archives are processed with `0.2` (or `"20%:0.3, 40%:0.2, 40%:0.1"`, etc.) ?

**Comment by danpovey** [comment]
If we want to support more general schedules, we'd want to use the same
format as for the dropout schedules, which is a piecewise linear function.
The format I suggested is a simpler version of that but compatible in
syntax.  But for now just do the simple thing... Vimal would need to be
involved to properly unify that code.

Dan

**Comment by hhadian** [comment]
OK. So in your format, does "0.2,0.1" mean a smooth change from 0.2 to 0.1 (like learning rate)?

**Comment by danpovey** [comment]
Yes, but I had in mind a linear change, not geometric like learning
rate.... actually an geometric interpolation might work better.  who knows.
 you could try both once we see if it's even helpful.

**Comment by hhadian** [comment]
OK.

**Comment by hhadian** [comment]
What do you want me to do with the output-xent learning rate factor which is a function of xent-regularize?:
```
learning_rate_factor=$(echo "print 0.5/$xent_regularize" | python)
output-layer name=output-xent dim=$num_targets learning-rate-factor=$learning_rate_factor max-change=1.5
```

**Comment by danpovey** [comment]
Use the average of the two.
